ca_qc_saint_jean_sur_richelieu/people.py

```diff
@@ -1,75 +1,43 @@
-from utils import CanadianScraper, CanadianPerson as Person, clean_string
-
-import re
+from utils import CanadianScraper, CanadianPerson as Person
+from urllib.parse import urljoin
 
 COUNCIL_PAGE = 'https://sjsr.ca/conseil-municipal/'
 
 
 class SaintJeanSurRichelieuPersonScraper(CanadianScraper):
     def scrape(self):
-        page = self.lxmlize(COUNCIL_PAGE)
+        page = self.lxmlize(COUNCIL_PAGE, encoding='utf-8')
 
-        councillors = page.xpath('//div[@class="article-content"]//td[@class="ms-rteTableOddCol-0"]')
-        yield self.scrape_mayor(councillors[0])
+        councillors = page.xpath('//div[contains(@class,"fl-module-content fl-node-content")]/div[@class="fl-rich-text"]')[1:]
         assert len(councillors), 'No councillors found'
-        for councillor in councillors[1:]:
-            if not councillor.xpath('.//a'):
-                continue
-
-            texts = [text for text in councillor.xpath('.//text()') if clean_string(text)]
-            name = texts[0]
-            district = texts[1]
-            url = councillor.xpath('.//a/@href')[0]
-            page = self.lxmlize(url)
-
-            p = Person(primary_org='legislature', name=name, district=district, role='Conseiller')
-            p.add_source(COUNCIL_PAGE)
-            p.add_source(url)
+        for councillor in councillors:
 
-            p.image = councillor.xpath('./preceding-sibling::td//img/@src')[-1]
+            name = councillor.xpath('.//a//text()')[0]
+            url = councillor.xpath('.//a//@href')[0]
 
-            contacts = page.xpath('.//td[@class="ms-rteTableOddCol-0"]//text()')
-            for contact in contacts:
-                if re.findall(r'[0-9]{4}', contact):
-                    phone = contact.strip().replace(' ', '-')
-                    p.add_contact('voice', phone, 'legislature')
-            get_links(p, page.xpath('.//td[@class="ms-rteTableOddCol-0"]')[0])
+            if (councillor.xpath('.//a/@href[contains(.,"maire")]')):
+                ward = 'Saint-Jean-sur-Richelieu'
+                role = 'Maire'
+            else:
+                role = 'Conseiller'
+                district = councillor.xpath('.//p[contains(.,"District")]/text()')
+                for i in district:
+                    sep = ','
+                    ward = i.split(sep, 1)[0]
 
-            email = self.get_email(page)
-            p.add_contact('email', email)
-            yield p
+            yield self.councillor_data(url, name, ward, role)
 
-    def scrape_mayor(self, div):
-        name = div.xpath('.//a')[0].text_content()
-        url = div.xpath('.//a/@href')[0]
-        page = self.lxmlize(url)
-        contact_url = page.xpath('//a[@title="Joindre le maire"]/@href')[0]
-        contact_page = self.lxmlize(contact_url)
+    def councillor_data(self, url, name, ward, role):
```

Comment:
Since this method is only called from one place, please remove it, and move its body into the main `parse` method.